pegasus/sites.v3/code.org/public/js/workshop-search.js

```diff
@@ -48,15 +55,61 @@ function loadWorkshops() {
               info_window.setContent(this.get('infowindow_content'));
               info_window.open(gmap, this);
             });
-        }
-        else {
+        } else {
           infowindow_content = compileHtml(workshop, false);
           // Extend existing marker.
           markers[hash].infowindow_content += infowindow_content;
         }
       }
     });
-  }).done(addGeocomplete).fail(displayQueryError);
+  }).fail(displayQueryError).always(addGeocomplete());
+}
+
+function processPdWorkshops(workshops) {
+  remainingWorkshops = workshops.length;
+  $.each(workshops, function (i, workshop) {
+    var address = workshop.location_address;
+
+    var geocoder = new google.maps.Geocoder();
+    geocoder.geocode( { 'address' : address }, function (results, status) {
+      if (status == google.maps.GeocoderStatus.OK) {
+        var location = results[0].geometry.location;
+        var lat = location.lat();
+        var lng = location.lng();
+        var hash = new google.maps.LatLng(lat, lng).toUrlValue();
+
+        var infowindow_content = '';
+
+        if (typeof markers[hash]==='undefined') {
+          infowindow_content = compileHtmlNew(workshop, true);
```

Comment:
might as well. I copied and modified this function from above. I'll take another quick pass at cleaning it up slightly.